engines/native/sandbox.go

```diff
@@ -195,20 +214,36 @@ func (s *sandbox) Abort() error {
 	s.resolve.Do(func() {
 		debug("Aborting sandbox")
 
-		// Kill sub-process
-		s.process.Kill()
-
-		// Halt all other sub-processes
-		err := system.KillByOwner(s.user)
-		if err != nil {
-			s.log.Error("Failed to kill all processes by owner, error: ", err)
+		// In case we didn't create a new user, killing
+		// the children processes is the only safe way
+		// to kill processes created by the task.
+		// We must kill the children after the parent
+		// to avoid the case of the parent creating new
+		// processes after we kill the descendents.
+		// On the other hand, we must kill children before
+		// the parent process is `waited`, because then
+		// the children process will be inherited by the init
+		// process in posix systems, and KillChildren won't
+		// work anymore.
+		// So, we run process.Kill() in a goroutine to
+		// "parallelize" the killing process.
```

Comment:
This sounds like a super racy solution...

So in `process_posix.go` we should only do `go p.waitForResult()` the first time `process.Wait()` is called...
Instead of starting to wait on the process in `StartProcess`.... Though after thinking I don't think that'll help as `sandbox.Abort()`  maybe called while `sandbox.WaitForResult()` is waiting. So we would need some wait to abort the pending wait operation.

Either way, it seems like this is platform specific information... it belongs in `system`.
So let's create a `system.KillProcessTree(s.process)` instead... Where `KillProcessTree` terminates a process and all it's children. Exactly, how we get that working and avoid races can be handled in `system` where the other `Wait` call is likely blocking in parallel.

---
Otherwise, like the idea of doing `system.KillProcessTree` followed by `KillByOwner(s.user)` if `CreateUser` is true...